samcli/lib/build/app_builder.py

```diff
@@ -502,6 +502,13 @@ def _build_layer(
         str
             Path to the location where built artifacts are available
         """
+
+        if layer_metadata and "BuildArchitecture" not in layer_metadata:
+            LOG.warning(
+                "WARNING: No BuildArchitecture specified in Layer `%s`" + " Metadata. Defaulting to x86_64.",
+                layer_name,
+            )
```

Comment:
I think we want to print this warning message if layer has `arm64` architecture as [`CompatibleArchitectures`](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-layerversion.html#sam-layerversion-compatiblearchitectures) but it doesn't have `BuildArchitecture` property

Reply:
There is [another warning ](https://github.com/aws/aws-sam-cli/blob/be0eb703c844db2498fbb6919b0c8c20997a4943/samcli/lib/providers/provider.py#L256-L261)in the LayerVersion constructor that should take care of this.

It would output
```
WARNING: Layer `Layer` has BuildArchitecture `x86_64`, which is not listed in CompatibleArchitectures. 
```

Reply:
This _may_ be slightly confusing if the customer doesn't know that x86 is the default? Maybe we can adjust the messaging here

Reply:
My concern with this one is; if I have a layer which is supposed to be built with `x86` I will keep getting these warning messages all the time, which can be annoying.

Reply:
Right. As discussed, going to restructure according to these rules:

There are two cases where we'd like to warn the customer
1. Compatible Architectures is only x86 (or not present) but Build Architecture is arm64
2. Build Architecture is x86 (or not present) but Compatible Architectures is only arm64